render/respond.go

```diff
@@ -0,0 +1,111 @@
+package render
+
+import (
+	"bytes"
+	"encoding/json"
+	"encoding/xml"
+	"net/http"
+	"reflect"
+
+	"golang.org/x/net/context"
+)
+
+var Respond = DefaultRespond
+
+func DefaultRespond(ctx context.Context, w http.ResponseWriter, v interface{}) {
```

Comment:
Ad. 1. Matters only for a streaming response of which we have 1 and that could be handled as a special case.

Ad. 2. Also an option

Ad. 3 This will only work until the 1st byte of the body is written, so it will not work anywhere in middleware chain in the code run after `next.ServeHTTP`